### PR TITLE
Remove dependency to UI, use RAII to handle resources

### PR DESCRIPTION
**Pull request by guilpier-code**
Fix issue #1658 

- [x] extract **main(...)** arguments processing from UI
- [x] Call **main(...)** arguments processing from the physical new location for each call
- [x] Make the arguments processing a class (or a wrapper) and use the RAII to avoid calling a destruction function
- [x] Correct the location of the arguments processing header (.h) to stick to what is done elsewhere (**args/antares/args**)

### CODE REVIEW

File: src/libs/antares/args/args_to_utf8.h
Comment (flomnes) [inline src/libs/antares/args/args_to_utf8.h:1]: Please create directory **antares/args** here and put this header file inside. 

See [this ADR](https://github.com/AntaresSimulatorTeam/Antares_Simulator/blob/develop/docs/Architecture_Decision_Records/CMake_design_choice.md)

See for example **src/libs/antares/date/antares/date/date.h**.

Comment (guilpier-code) [reply src/libs/antares/args/args_to_utf8.h:1]: Done

File: src/libs/antares/args/args_to_utf8.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+{
+    wchar_t** wargv = CommandLineToArgvW(GetCommandLineW(), &argc);
+    char** argvUTF8 = (char**)malloc(argc * sizeof(char*));
+    for (int i = 0; i != argc; ++i)
+    {
+        const uint len = (uint)wcslen(wargv[i]);
+        const uint newLen = WideCharToMultiByte(CP_UTF8, 0, wargv[i], len, NULL, 0, NULL, NULL);
+        argvUTF8[i] = (char*)malloc((newLen + 1) * sizeof(char));
+        memset(argvUTF8[i], 0, (newLen + 1) * sizeof(char));
+        WideCharToMultiByte(CP_UTF8, 0, wargv[i], len, argvUTF8[i], newLen, NULL, NULL);
```
Comment (flomnes) [inline src/libs/antares/args/args_to_utf8.cpp]: I think you could simplify a bit
```cpp
char** AntaresGetUTF8Arguments(int argc, char**)
{
#ifdef YUNI_OS_WINDOWS
// windows implementation here
#else
// non-windows implementation here
#endif
}
```

**Comment by flomnes** [comment]
For the RAII topic, you have (at least) two options

- Make a C++ class that owns a `char**`
- Make this function operate on and return `std::vector<std::string>` or `std::vector<std::wstring>` and let `std::vector` manage it's own memory

**Comment by guilpier-code** [comment]
> For the RAII topic, you have (at least) two options
> 
> * Make a C++ class that owns a `char**`
> * Make this function operate on and return `std::vector<std::string>` or `std::vector<std::wstring>` and let `std::vector` manage it's own memory

Ok, knowing that the second change is more invasive than the first.

File: src/libs/antares/args/args_to_utf8.cpp (hunk 1 above)
Comment (guilpier-code) [reply src/libs/antares/args/args_to_utf8.cpp]: Done, even if a class now replaces the 2 functions

File: src/libs/antares/args/args_to_utf8.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN 1
+#endif
+#include <windows.h>
+#include <shellapi.h>
+#include <stdlib.h>
+#include <stdio.h>
+#endif
+
+IntoUTF8ArgsTranslator::IntoUTF8ArgsTranslator(int argc, char**& argv)
```
Comment (flomnes) [inline src/libs/antares/args/args_to_utf8.cpp]:
```suggestion
IntoUTF8ArgsTranslator::IntoUTF8ArgsTranslator(int argc, char** argv)
```

File: src/libs/antares/args/include/antares/args/args_to_utf8.h [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+class IntoUTF8ArgsTranslator
+{
+public:
+	IntoUTF8ArgsTranslator(int argc, char**& argv);
```
Comment (flomnes) [inline src/libs/antares/args/include/antares/args/args_to_utf8.h]:
```suggestion
	IntoUTF8ArgsTranslator(int argc, char** argv);
```

File: src/libs/antares/args/include/antares/args/args_to_utf8.h [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+	~IntoUTF8ArgsTranslator();
+private:
+	int argc_ = 0;
+	char**& argv_;
```
Comment (flomnes) [inline src/libs/antares/args/include/antares/args/args_to_utf8.h]:
```suggestion
	char** argv_;
```

File: src/libs/antares/args/args_to_utf8.cpp [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+#ifdef YUNI_OS_WINDOWS
+#include <string.h>
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN 1
+#endif
+#include <windows.h>
+#include <shellapi.h>
+#include <stdlib.h>
+#include <stdio.h>
```
Comment (flomnes) [inline src/libs/antares/args/args_to_utf8.cpp]:
```suggestion
```

Use `new` & `delete`

File: src/libs/antares/args/args_to_utf8.cpp [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+#include "antares/args/args_to_utf8.h"
+
+
+#ifdef YUNI_OS_WINDOWS
+#include <string.h>
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN 1
+#endif
+#include <windows.h>
+#include <shellapi.h>
```
Comment (flomnes) [inline src/libs/antares/args/args_to_utf8.cpp:12]: It seems that these headers only make sense for Windows, right ?

**Review by flomnes — CHANGES_REQUESTED** [review]
I suggest adding a convert method that returns a new `argv` as a `char**`

**Review by flomnes — APPROVED** [review]

**Review by flomnes — APPROVED** [review]

**Pull request merged**